pkg/configapi/types.go

```diff
@@ -0,0 +1,336 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package configapi
```

Comment:
Instead of "configapi", I suggest "clusterconfig".

It also pains me to add another top-level package, though I don't have a suggested alternative at the moment. Unless we break up the config by component, in which case we could push them down to component-specific directories. Do all the types actually need to go into a single file/directory? It doesn't look like there are shared types at the moment.